graylog2-server/src/main/java/org/graylog2/rest/resources/system/indices/RetentionStrategyResource.java

```diff
@@ -112,6 +117,20 @@ private RetentionStrategyDescription getRetentionStrategyDescription(@ApiParam(n
             throw new InternalServerErrorException("Couldn't generate JSON schema for retention strategy " + strategyName, e);
         }
 
-        return RetentionStrategyDescription.create(strategyName, defaultConfig, visitor.finalSchema());
+        JsonSchema jsonSchema = visitor.finalSchema();
+        removeDeactivatedStrategiesActions(jsonSchema);
+
+        return RetentionStrategyDescription.create(strategyName, defaultConfig, jsonSchema);
+    }
+
+    private void removeDeactivatedStrategiesActions(JsonSchema schema) {
+        Map<String, JsonSchema> properties = ((ObjectSchema) schema).getProperties();
+        JsonSchema indexAction = properties.get("index_action");
+
+        if (Objects.nonNull(indexAction)) {
+            Set<String> actionEnums = ((StringSchema) indexAction).getEnums();
+            Set<String> disabledRetentionStrategies = elasticsearchConfiguration.getDisabledRetentionStrategies();
+            disabledRetentionStrategies.stream().map(s -> s.toUpperCase(Locale.ENGLISH)).toList().forEach(actionEnums::remove);
+        }
```

Comment:
Glad you found a solution :+1: 
However, this only works for hiding it in the frontend. the API would still accept every `index_action`.
I think we need to add a pluggable validation method to the `RetentionStrategy` interface.
And implement the validation in the `ArchiveRetentionStrategy`

Reply:
added validation to the config interface